src/plugins/procmon/procmon.cpp

```diff
@@ -371,6 +386,55 @@ static event_response_t process_create_ex_return_hook(drakvuf_t drakvuf, drakvuf
     return VMI_EVENT_RESPONSE_NONE;
 }
 
+static event_response_t process_create_return_hook(drakvuf_t drakvuf, drakvuf_trap_info_t* info)
+{
+    auto plugin = get_trap_plugin<procmon>(info);
+    auto params = get_trap_params<process_create_result_t>(info);
+
+    if (!params->verify_result_call_params(drakvuf, info))
+        return VMI_EVENT_RESPONSE_NONE;
+
+    addr_t process_handle_addr = params->process_handle_addr;
+    reg_t status = info->regs->rax;
+
+    ACCESS_CONTEXT(ctx,
+        .translate_mechanism = VMI_TM_PROCESS_DTB,
+        .dtb = info->regs->cr3,
+        .addr = process_handle_addr,
+    );
+
+    addr_t process_handle;
+    vmi_instance_t vmi = drakvuf_lock_and_get_vmi(drakvuf);
+    if (VMI_FAILURE == vmi_read_addr(vmi, &ctx, &process_handle))
+        process_handle = 0;
+
+    drakvuf_release_vmi(drakvuf);
+
+    vmi_pid_t new_pid;
+    if (!drakvuf_get_pid_from_handle(drakvuf, info, process_handle, &new_pid))
+        new_pid = 0;
+
+    if (strstr(info->attached_proc_data.name,"notepad"))
+      // somehow stop this run
```

Comment:
return hook is called when the system call was already executed, thus you will not be able to block anything on this level, this has to be moved to `create_process_hook` (and use different method to detect if we are trying to launch notepad or not)

Reply:
Oh, I was confused about this on if I should be terminating it after launch or block the call. But, I haven't really found any other methods for finding the process name before callbacks. Will try searching more